aptos-move/block-executor/src/executor.rs

```diff
@@ -54,12 +56,123 @@ use std::{
     sync::{atomic::AtomicU32, Arc},
 };
 
+pub struct ProcessedAccumulator<T: Transaction> {
+    block_gas_limit_type: BlockGasLimitType,
+    accumulated_effective_block_gas: u64,
+    accumulated_fee_statement: FeeStatement,
+    txn_fee_statements: Vec<FeeStatement>,
+    txn_read_write_summaries: Vec<ReadWriteSummary<T>>,
+}
+
+impl<T: Transaction> ProcessedAccumulator<T> {
+    pub fn new(block_gas_limit_type: BlockGasLimitType, init_size: usize) -> Self {
+        Self {
+            block_gas_limit_type,
+            accumulated_effective_block_gas: 0,
+            accumulated_fee_statement: FeeStatement::zero(),
+            txn_fee_statements: Vec::with_capacity(init_size),
+            txn_read_write_summaries: Vec::with_capacity(init_size),
+        }
+    }
+
+    pub fn accumulate_fee_statement(
+        &mut self,
+        fee_statement: FeeStatement,
+        txn_read_write_summary: ReadWriteSummary<T>,
+    ) {
+        self.accumulated_fee_statement
+            .add_fee_statement(&fee_statement);
+        self.txn_fee_statements.push(fee_statement);
+        self.txn_read_write_summaries.push(txn_read_write_summary);
+
+        let conflict_multiplier = if let Some(conflict_overlap_length) =
+            self.block_gas_limit_type.conflict_overlap_window()
+        {
+            self.compute_conflict_multiplier(conflict_overlap_length as usize)
+        } else {
+            1
+        };
+
+        // println!("[{}] Multiplier {}, with read/write summary: {:?}", self.txn_read_write_summaries.len() - 1, conflict_multiplier, self.txn_read_write_summaries[self.txn_read_write_summaries.len() - 1]);
+
+        // When the accumulated execution and io gas of the committed txns exceeds
+        // PER_BLOCK_GAS_LIMIT, early halt BlockSTM. Storage gas does not count towards
+        // the per block gas limit, as we measure execution related cost here.
+        self.accumulated_effective_block_gas += conflict_multiplier
```

Comment:
so, possible to add output sizing info to the effective gas? or do we add a second limit?

Reply:
I'm gonna add a second limit tomorrow

Reply:
Thanks boss!